training/training/core/dataset.py

```diff
@@ -98,3 +102,64 @@ def getCategoryList(self) -> list[str]:
         if self._category_list is None:
             raise Exception("Category list not available")
         return self._category_list
+
+
+class CustomDatasetCreator(TrainTestDatasetCreator):
+    """Pulls user-uploaded dataset from S3 bucket and converts it to readable format"""
+
+    def __init__(
+        self,
+        X: pd.DataFrame,
+        y: pd.Series,
+        test_size: float,
+        shuffle: bool,
+        category_list: Optional[list[str]],
+    ) -> None:
+        super().__init__()
+        self._category_list = category_list
+        self._X_train, self._X_test, self._y_train, self._y_test = cast(
+            tuple[pd.DataFrame, pd.DataFrame, pd.Series, pd.Series],
+            train_test_split(X, y, test_size=test_size, shuffle=shuffle),
+        )
+
+    @classmethod
+    def read_s3(
+        cls,
+        uid: str,
+        name: str,
+        test_size: float,
+        target_name: str,
+        shuffle: bool = True,
+    ):
+        s3 = boto3.client("s3")
+        obj = s3.get_object(Bucket="dlp-upload-bucket", Key=f"{uid}/tabular/{name}")
```

Comment:
We should allow models other than tabular to access the uploaded datasets but this is okay for now

Reply:
TODO: add error handling if such a directory doesn't exist